pkg/comp-functions/runtime/naming.go

```diff
@@ -0,0 +1,82 @@
+package runtime
+
+import (
+	"regexp"
+	"strings"
+
+	"github.com/crossplane/function-sdk-go/resource/composed"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// escapeK8sNames will figure out what naming scheme applies to the given object
+// and tries to find the correct naming scheme for it.
+// Then it will escape the name accordingly.
+func escapeK8sNames(obj client.Object) {
+	kind, _, err := composed.Scheme.ObjectKinds(obj)
+	if err != nil {
+		obj.SetName(escapeDNS1123(obj.GetName(), false))
+	}
+
+	switch kind[0].Kind {
+	case "Pod":
+		obj.SetName(escapeDNS1123Label(obj.GetName()))
+	case "Namespace":
+		obj.SetName(escapeDNS1123Label(obj.GetName()))
+	case "Service":
+		obj.SetName(escapeDNS1123Label(obj.GetName()))
+	case "Role":
+		obj.SetName(escapeDNS1123(obj.GetName(), true))
+	case "ClusterRole":
+		obj.SetName(escapeDNS1123(obj.GetName(), true))
+	case "RoleBinding":
+		obj.SetName(escapeDNS1123(obj.GetName(), true))
+	case "ClusterRoleBinding":
+		obj.SetName(escapeDNS1123(obj.GetName(), true))
+	default:
+		obj.SetName(escapeDNS1123(obj.GetName(), false))
+	}
+}
+
+// escapeDNS1123Label does the same as escapeDNS1123 but also limit to 63 chars
+func escapeDNS1123Label(name string) string {
+	name = escapeDNS1123(name, false)
+	if len(name) > 63 {
+		name = name[:63]
```

Comment:
This is somehow Ok imho, there is a danger of overriding values if too long labels are truncated instead of returning an error:
```
app.kubernetes.io/this-is-a-very-long-label-key-to-reach-the-limit-user1111: "user1111"
app.kubernetes.io/this-is-a-very-long-label-key-to-reach-the-limit-user2222: "user2222"
```

if we truncate, we'll end up with only one label of value:
```
app.kubernetes.io/this-is-a-very-long-label-key-to-reach-the-l: user2222
```
competely losing logic

Reply:
I though about the same, I could introduce some randomness at the. Like what a deployment does.

BTW, you're confusing what label means in this context. Here label refers to a DNS label, which is RFC1123 and specifies what's allowed in a kubernetes object name, see https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#dns-label-names. We're escaping `metadata.name` here, not object labels, although labels probably also have some constraints to them I guess.